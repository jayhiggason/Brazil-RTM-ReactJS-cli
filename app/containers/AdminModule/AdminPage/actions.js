/*
 *
 * AdminPage actions
 *
 */

import {
    ADMIN_GET_REQUESTED_USERS_FETCH,
    ADMIN_GET_REQUESTED_USERS_FETCH_FAIL,
    ADMIN_GET_REQUESTED_USERS_FETCH_SUCCESS,
    ADMIN_PAGE_HANDLE_REQUEST,
    ADMIN_PAGE_HANDLE_TABS,
    DEFAULT_ACTION
} from "./constants";

export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

export function adminGetRequestedUsersFetch() {
    return {
        type: ADMIN_GET_REQUESTED_USERS_FETCH,
    };
}

export function adminGetRequestedUsersFetchSuccess(data) {
    return {
        type: ADMIN_GET_REQUESTED_USERS_FETCH_SUCCESS,
        data
    };
}

export function adminGetRequestedUsersFetchFail() {
    return {
        type: ADMIN_GET_REQUESTED_USERS_FETCH_FAIL,
    };
}

export function adminHandleRequest(userRequest) {
    return {
        type: ADMIN_PAGE_HANDLE_REQUEST,
        userRequest
    };
}

export function adminHandleTabs(tab) {
    return {
        type: ADMIN_PAGE_HANDLE_TABS,
        tab
    };
}
