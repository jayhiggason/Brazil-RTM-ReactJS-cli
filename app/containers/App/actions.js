import {
    CHANGE_DATA_LOAD_STATUS,
    CHANGE_PANEL,
    CLOSE_MAIN_DRAWER, DATA_EXCEPTION_ALERT,
    FILTER_DATA_FAIL,
    FILTER_DATA_FETCH,
    FILTER_DATA_SUCCESS,
    HANDLE_EXPLAIN_MODULE,
    ON_CHANGE_PI_RISK_CONTROL_PANEL_DATA,
    ON_CHANGE_PI_SHELF_LIFE_RISK_CONTROL_PANEL_DATA,
    ON_CLOSING_PI_GLOBAL_SETTINGS,
    ON_OPENING_PI_GLOBAL_SETTINGS,
    OPEN_MAIN_DRAWER,
    PI_FILTER_SET,
    PI_RISK_CONTROL_ON_CHANGE_DATA_LOADED_CHANGE_STATUS,
    PI_SHELF_LIFE_RISK_CONTROL_ON_CHANGE_DATA_LOADED_CHANGE_STATUS,
    PIN_MY_PAGE,
    PIN_MY_VIEW,
    PIN_MY_VIEW_CLOSE_DIALOG,
    PIN_MY_VIEW_CONFIRM,
    RESET_DEEP_DIVE,
    RISK_CONTROL_PANEL_SAVE_RISKS,
    RISK_CONTROL_PANEL_SAVE_RISKS_COMPLETED,
    RISK_CONTROL_PANEL_SAVE_SHELF_LIFE_RISKS,
    RISK_CONTROL_PANEL_SAVE_SHELF_LIFE_RISKS_COMPLETED,
    SCMOS_BRAND_TECH_RISK_CLASSIFICATION_DATA_FETCH,
    SCMOS_BRAND_TECH_RISK_CLASSIFICATION_DATA_FETCH_FAIL,
    SCMOS_BRAND_TECH_RISK_CLASSIFICATION_DATA_FETCH_SUCCESS,
    SELECTED_FILTERS_APPLY,
    SHOW_APP_ALERT,
    STORE_FILTER_DATA_FAIL,
    STORE_FILTER_DATA_FETCH,
    STORE_FILTER_DATA_SUCCESS,
    UNPIN_MY_VIEW
} from "./constants";


export function openMainDrawer() {
    return {
        type: OPEN_MAIN_DRAWER
    };
}

export function closeMainDrawer() {
    return {
        type: CLOSE_MAIN_DRAWER
    };
}

export function changePanel(data) {
    return {
        type: CHANGE_PANEL,
        data
    }
}


export function setPIFilter(data) {
    return {
        type: PI_FILTER_SET,
        data
    }
}

export function onOpeningPIGlobalSettings() {
    return {
        type: ON_OPENING_PI_GLOBAL_SETTINGS,
    };
}

export function onClosingPIGlobalSettings() {
    return {
        type: ON_CLOSING_PI_GLOBAL_SETTINGS,
    };
}

export function PIRiskControlPanelDataOnChange(data) {
    return {
        type: ON_CHANGE_PI_RISK_CONTROL_PANEL_DATA,
        data
    };
}

export function PIShelfLifeRiskControlPanelDataOnChange(data) {
    return {
        type: ON_CHANGE_PI_SHELF_LIFE_RISK_CONTROL_PANEL_DATA,
        data
    };
}

export function PIRiskControlPanelOnChangeDataLoadedChangeStatus(data) {
    return {
        type: PI_RISK_CONTROL_ON_CHANGE_DATA_LOADED_CHANGE_STATUS,
        data
    };
}

export function PIShelfLifeRiskControlPanelOnChangeDataLoadedChangeStatus(data) {
    return {
        type: PI_SHELF_LIFE_RISK_CONTROL_ON_CHANGE_DATA_LOADED_CHANGE_STATUS,
        data
    };
}

export function riskControlPanelSaveRisks() {
    return {
        type: RISK_CONTROL_PANEL_SAVE_RISKS
    }
}

export function showAppAlert(alerts) {
    return {
        type: SHOW_APP_ALERT,
        alerts
    }
}

export function handleExplainModule(state) {
    return {
        type: HANDLE_EXPLAIN_MODULE,
        state
    }
}

export function scmosBrandTechRiskClassificationDataFetch() {
    return {
        type: SCMOS_BRAND_TECH_RISK_CLASSIFICATION_DATA_FETCH,
    }
}

export function scmosBrandTechRiskClassificationDataFetchSuccess(data) {
    return {
        type: SCMOS_BRAND_TECH_RISK_CLASSIFICATION_DATA_FETCH_SUCCESS,
        data
    }
}

export function scmosBrandTechRiskClassificationDataFetchFail() {
    return {
        type: SCMOS_BRAND_TECH_RISK_CLASSIFICATION_DATA_FETCH_FAIL,
    }
}


export function riskControlPanelSaveShelfLifeRisks() {
    return {
        type: RISK_CONTROL_PANEL_SAVE_SHELF_LIFE_RISKS,
    }
}

export function riskControlPanelSaveShelfLifeRisksCompleted() {
    return {
        type: RISK_CONTROL_PANEL_SAVE_SHELF_LIFE_RISKS_COMPLETED,
    }
}

export function riskControlPanelSaveRisksCompleted() {
    return {
        type: RISK_CONTROL_PANEL_SAVE_RISKS_COMPLETED,
    }
}


// rtm filters
export function filterDataFetch(data) {
    return {
        type: FILTER_DATA_FETCH,
        data
    };
}

export function filterDataFail() {
    return {
        type: FILTER_DATA_FAIL
    };
}

export function filterDataSuccess(data) {
    return {
        type: FILTER_DATA_SUCCESS,
        data
    };
}

export function selectedFilterOnChange(data) {
    return {
        type: SELECTED_FILTERS_APPLY,
        data
    }
}

//rtm store filter
export function storeFilterDataFetch() {
    return {
        type: STORE_FILTER_DATA_FETCH
    };
}

export function storeFilterDataFailed() {
    return {
        type: STORE_FILTER_DATA_FAIL
    };
}

export function storeFilterDataSuccess(data) {
    return {
        type: STORE_FILTER_DATA_SUCCESS,
        data
    };
}

export function pinMyPage(data) {
    return {
        type: PIN_MY_PAGE,
        data
    };
}

export function changeDataLoadStatus(data) {
    return {
        type: CHANGE_DATA_LOAD_STATUS,
        data
    };
}

export function pinMyView(data) {
    return {
        type: PIN_MY_VIEW,
        data
    };
}

export function unpinMyView(data) {
    return {
        type: UNPIN_MY_VIEW,
        data
    };
}

export function pinMyViewConfirm(data) {
    return {
        type: PIN_MY_VIEW_CONFIRM,
        data
    };
}

export function pinMyViewCloseDialog() {
    return {
        type: PIN_MY_VIEW_CLOSE_DIALOG,
    };
}

export function resetDeepDive(data) {
    return {
        type: RESET_DEEP_DIVE,
        data
    };
}

//data exception handling

export function dataExceptionHandlingDialog() {
    return {
        type: DATA_EXCEPTION_ALERT,
    };
}
