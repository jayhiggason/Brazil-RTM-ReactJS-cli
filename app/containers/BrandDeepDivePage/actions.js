/*
 *
 * BrandDeepDivePage actions
 *
 */

import {
    CARDS_DATA_FAIL,
    CARDS_DATA_FETCH,
    CARDS_DATA_SUCCESS,
    CATEGORY_COMPARISON_FILTER_FAIL,
    CATEGORY_COMPARISON_FILTER_FETCH,
    CATEGORY_COMPARISON_FILTER_SUCCESS,
    CATEGORY_FILTER_FAIL,
    CATEGORY_FILTER_FETCH,
    CATEGORY_FILTER_SUCCESS,
    DEFAULT_ACTION,
    INVOICE_TOGGLE,
    ITEM_HEATMAP_BUTTON,
    ITEM_HEATMAP_FAIL,
    ITEM_HEATMAP_FETCH,
    ITEM_HEATMAP_FILTER,
    ITEM_HEATMAP_PARTITION_FILTER,
    ITEM_HEATMAP_SUCCESS,
    ITEM_HEATMAP_TIME_FILTER,
    PRODUCT_BUTTON_VS,
    PRODUCT_SKU_TREND_FILTER_FAIL,
    PRODUCT_SKU_TREND_FILTER_FETCH,
    PRODUCT_SKU_TREND_FILTER_SELECTED_FILTER,
    PRODUCT_SKU_TREND_FILTER_SUCCESS,
    PRODUCT_TREND_FAIL,
    PRODUCT_TREND_FETCH,
    PRODUCT_TREND_FILTER,
    PRODUCT_TREND_SUCCESS,
    PRODUCT_UOM_FILTER,
    SELECTED_CATEGORY,
    SELECTED_CATEGORY_COMPARISON,
    SKU_TABLE_FAIL,
    SKU_TABLE_FETCH,
    SKU_TABLE_FILTER,
    SKU_TABLE_SUCCESS,
    TOGGLE_CARD,
    TRENDS_LINE_FAIL,
    TRENDS_LINE_FETCH,
    TRENDS_LINE_SUCCESS,
    UNITS_TOGGLE
} from "./constants";

export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

// cards data

export function cardDataFetch() {
    return {
        type: CARDS_DATA_FETCH
    };
}

export function cardDataSuccess(data) {
    return {
        type: CARDS_DATA_SUCCESS,
        data
    };
}

export function cardDataFail() {
    return {
        type: CARDS_DATA_FAIL
    };
}

export function invoiceToggle(data) {
    return {
        type: INVOICE_TOGGLE,
        data
    };
}

export function unitsToggle(data) {
    return {
        type: UNITS_TOGGLE,
        data
    };
}

export function toggleCard(data) {
    return {
        type: TOGGLE_CARD,
        data
    };
}

// see trends filters

export function categoryFilterFetch() {
    return {
        type: CATEGORY_FILTER_FETCH
    };
}

export function categoryFilterSuccess(data) {
    return {
        type: CATEGORY_FILTER_SUCCESS,
        data
    };
}

export function categoryFilterFail() {
    return {
        type: CATEGORY_FILTER_FAIL
    };
}

export function categoryOnChange(data) {
    return {
        type: SELECTED_CATEGORY,
        data
    };
}

export function categoryComparisonFilterFetch() {
    return {
        type: CATEGORY_COMPARISON_FILTER_FETCH
    };
}

export function categoryComparisonFilterSuccess(data) {
    return {
        type: CATEGORY_COMPARISON_FILTER_SUCCESS,
        data
    };
}

export function categoryComparisonFilterFail() {
    return {
        type: CATEGORY_COMPARISON_FILTER_FAIL
    };
}

export function categoryComparisonOnChange(data) {
    return {
        type: SELECTED_CATEGORY_COMPARISON,
        data
    };
}

// see trends line

export function seeTrendLineFetch(data) {
    return {
        type: TRENDS_LINE_FETCH,
        data
    };
}

export function seeTrendLineSuccess(data) {
    return {
        type: TRENDS_LINE_SUCCESS,
        data
    };
}

export function seeTrendLineFail() {
    return {
        type: TRENDS_LINE_FAIL
    };
}


// PRODUCT/SKU graph

export function productTrendFetch() {
    return {

        type: PRODUCT_TREND_FETCH
    };
}

export function productTrendFail() {
    return {
        type: PRODUCT_TREND_FAIL
    };
}

export function productTrendSuccess(data) {
    return {
        type: PRODUCT_TREND_SUCCESS,
        data
    };
}

//Product trend buttons
export function productTrendClickOnChange(data) {
    return {
        type: PRODUCT_BUTTON_VS,
        data
    };
}


// PRODUCT TREND filter
export function productTrendFilterOnChange(data) {
    return {
        type: PRODUCT_TREND_FILTER,
        data
    };
}

//product uom filter
export function productTrendUomFilterOnChange(data) {
    return {
        type: PRODUCT_UOM_FILTER,
        data
    };
}

//product sku trend filter
export function productSkuTrendFilterFetch() {
    return {
        type: PRODUCT_SKU_TREND_FILTER_FETCH
    };
}

export function productSkuTrendFilterSuccess(data) {
    return {
        type: PRODUCT_SKU_TREND_FILTER_SUCCESS,
        data
    };
}

export function productSkuTrendFilterFail() {
    return {
        type: PRODUCT_SKU_TREND_FILTER_FAIL
    };
}

export function productSkuTrendFilterOnChange(data) {
    return {
        type: PRODUCT_SKU_TREND_FILTER_SELECTED_FILTER,
        data
    };
}

// ITEM heat map

export function itemHeatFetch() {
    return {
        type: ITEM_HEATMAP_FETCH
    };
}

export function itemHeatFail() {
    return {
        type: ITEM_HEATMAP_FAIL
    };
}

export function itemHeatSuccess(data) {
    return {
        type: ITEM_HEATMAP_SUCCESS,
        data
    };
}

// ITEM HEATMAP button
export function itemHeatmapButtonOnChange(data) {
    return {
        type: ITEM_HEATMAP_BUTTON,
        data
    };
}

// ITEM HEATMAP filter
export function itemFilterHeatmapOnChange(data) {
    return {
        type: ITEM_HEATMAP_FILTER,
        data
    };
}

//item heatmap time filter
export function itemTimeFilterHeatmapOnChange(data) {
    return {
        type: ITEM_HEATMAP_TIME_FILTER,
        data

    }
}

// item heatmap partition filter
export function itemPartitionFilterHeatmapOnChange(data) {
    return {
        type: ITEM_HEATMAP_PARTITION_FILTER,
        data

    }
}

//SKU TABLE

export function skuTableFetch() {
    return {
        type: SKU_TABLE_FETCH
    };
}

export function skuTableSuccess(data) {
    return {
        type: SKU_TABLE_SUCCESS,
        data
    };
}

export function skuTableFail() {
    return {
        type: SKU_TABLE_FAIL
    };
}


export function skuFilterOnChange(data) {
    return {
        type: SKU_TABLE_FILTER,
        data
    };
}

