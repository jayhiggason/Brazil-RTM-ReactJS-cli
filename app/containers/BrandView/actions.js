/*
 *
 * BrandView actions
 *
 */

import {
    BRAND_BUTTON_VS,
    BRAND_HEATMAP_BUTTON,
    BRAND_HEATMAP_FAIL,
    BRAND_HEATMAP_FETCH,
    BRAND_HEATMAP_SUCCESS,
    BRAND_HEATMAP_TIME_FILTER,
    BRAND_HEATMAP_TOP,
    BRAND_PARTITION_HEATMAP_FILTER,
    BRAND_TREND_CLICK_ON_CHANGE,
    BRAND_TREND_FAIL,
    BRAND_TREND_FETCH,
    BRAND_TREND_SUCCESS,
    BRAND_TREND_TOP,
    BRAND_UOM_BUTTON,
    DEFAULT_ACTION
} from "./constants";

export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

// brandTrend graph

export function brandTrendFetch() {
    return {
        type: BRAND_TREND_FETCH
    };
}

export function brandTrendFail() {
    return {
        type: BRAND_TREND_FAIL
    };
}

export function brandTrendSuccess(data) {
    return {
        type: BRAND_TREND_SUCCESS,
        data
    };
}

//BRAND TREND CLICK
export function brandTrendClickOnChange(data) {
    return {
        type: BRAND_TREND_CLICK_ON_CHANGE,
        data
    };
}

// button on change
export function brandButtonVsOnChange(data) {
    return {
        type: BRAND_BUTTON_VS,
        data
    };
}

// top filter
export function brandTopFilterOnChange(data) {
    return {
        type: BRAND_TREND_TOP,
        data
    };
}

//brand uom button
export function brandUomButtonOnChange(data) {
    return {
        type: BRAND_UOM_BUTTON,
        data
    };
}

// brand heatmap

export function brandHeatFetch() {
    return {
        type: BRAND_HEATMAP_FETCH
    };
}

export function brandHeatFail() {
    return {
        type: BRAND_HEATMAP_FAIL
    };
}

export function brandHeatSuccess(data) {
    return {
        type: BRAND_HEATMAP_SUCCESS,
        data
    };
}

// button on change
export function brandHeatmapButtonOnChange(data) {
    return {
        type: BRAND_HEATMAP_BUTTON,
        data
    };
}

// top filter
export function topFilterHeatmapOnChange(data) {
    return {
        type: BRAND_HEATMAP_TOP,
        data
    };
}

//heatmap time filter
export function brandHeatmapTimeFilterOnChange(data) {
    return {
        type: BRAND_HEATMAP_TIME_FILTER,
        data
    };

}

//heatmap partition filter
export function brandHeatmapPartitionFilterOnChange(data) {
    return {
        type: BRAND_PARTITION_HEATMAP_FILTER,
        data
    };

}
