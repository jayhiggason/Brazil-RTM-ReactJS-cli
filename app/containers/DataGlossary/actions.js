/*
 *
 * DataGlossary actions
 *
 */

import {
    DATA_GLOSSARY_CRUD_OPERATION,
    DATA_GLOSSARY_CRUD_OPERATION_COMPLETED,
    DATA_GLOSSARY_DATA_FETCH,
    DATA_GLOSSARY_DATA_FETCH_FAIL,
    DATA_GLOSSARY_DATA_FETCH_SUCCESS,
    DEFAULT_ACTION
} from "./constants";

export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

export function dataGlossaryDataFetch() {
    return {
        type: DATA_GLOSSARY_DATA_FETCH
    };
}

export function dataGlossaryDataFetchSuccess(data) {
    return {
        type: DATA_GLOSSARY_DATA_FETCH_SUCCESS,
        data
    }
}

export function dataGlossaryDataFetchFail() {
    return {
        type: DATA_GLOSSARY_DATA_FETCH_FAIL
    }
}

export function dataGlossaryCrudOperation(data) {
    return {
        type: DATA_GLOSSARY_CRUD_OPERATION,
        data
    }
}

export function dataGlossaryCrudOperationCompleted() {
    return {
        type: DATA_GLOSSARY_CRUD_OPERATION_COMPLETED
    }
}
