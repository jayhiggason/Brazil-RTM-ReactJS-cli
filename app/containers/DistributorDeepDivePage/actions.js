/*
 *
 * DistributorDeepDivePage actions
 *
 */

import {
    CHANNEL_PERFORMANCE_BUTTON,
    CHANNEL_PERFORMANCE_CLICK_ON_CHANGE,
    CHANNEL_PERFORMANCE_FAIL,
    CHANNEL_PERFORMANCE_FETCH,
    CHANNEL_PERFORMANCE_FILTER,
    CHANNEL_PERFORMANCE_SUCCESS,
    DEFAULT_ACTION,
    DISTRIBUTOR_CARDS_DATA_FAIL,
    DISTRIBUTOR_CARDS_DATA_FETCH,
    DISTRIBUTOR_CARDS_DATA_SUCCESS,
    DISTRIBUTOR_SALES_PERFORMANCE_BUTTON_VS,
    DISTRIBUTOR_SALES_PERFORMANCE_FAIL,
    DISTRIBUTOR_SALES_PERFORMANCE_FETCH,
    DISTRIBUTOR_SALES_PERFORMANCE_KPI_BUTTON,
    DISTRIBUTOR_SALES_PERFORMANCE_SUCCESS,
    PRODUCT_PERFORMANCE_BUTTON,
    PRODUCT_PERFORMANCE_FAIL,
    PRODUCT_PERFORMANCE_FETCH,
    PRODUCT_PERFORMANCE_SUCCESS, STORE_PERFORMANCE_BUTTON,
    STORE_PERFORMANCE_FAIL,
    STORE_PERFORMANCE_FETCH,
    STORE_PERFORMANCE_FILTER,
    STORE_PERFORMANCE_SUCCESS, STORE_TABLE_BUTTON,
    STORE_TABLE_FAIL,
    STORE_TABLE_FETCH,
    STORE_TABLE_SUCCESS,
    TOGGLE_CARD,
    VISIT_SALES_BUBBLE_FAIL,
    VISIT_SALES_BUBBLE_FETCH, VISIT_SALES_BUBBLE_FILTER,
    VISIT_SALES_BUBBLE_SUCCESS
} from "./constants";

export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

//dist chart
export function distributorCardsFetch() {
    return {
        type: DISTRIBUTOR_CARDS_DATA_FETCH
    };
}

export function distributorCardsFail() {
    return {
        type: DISTRIBUTOR_CARDS_DATA_FAIL
    };
}

export function distributorCardsSuccess(data) {
    return {
        type: DISTRIBUTOR_CARDS_DATA_SUCCESS,
        data
    };
}

export function toggleCard(data) {
    return {
        type: TOGGLE_CARD,
        data
    };
}

//visit vs sales chart
export function visitSalesBubbleFetch() {
    return {
        type: VISIT_SALES_BUBBLE_FETCH
    };
}

export function visitSalesBubbleFail() {
    return {
        type: VISIT_SALES_BUBBLE_FAIL
    };
}

export function visitSalesBubbleSuccess(data) {
    return {
        type: VISIT_SALES_BUBBLE_SUCCESS,
        data
    };
}
export function visitSalesBubbleFilterOnChange(data) {
    return {
        type: VISIT_SALES_BUBBLE_FILTER,
        data
    };
}


// dist sales performance graph

export function distributorSalesPerformanceFetch() {
    return {
        type: DISTRIBUTOR_SALES_PERFORMANCE_FETCH
    };
}

export function distributorSalesPerformanceFail() {
    return {
        type: DISTRIBUTOR_SALES_PERFORMANCE_FAIL
    };
}

export function distributorSalesPerformanceSuccess(data) {
    return {
        type: DISTRIBUTOR_SALES_PERFORMANCE_SUCCESS,
        data
    };
}

// button on change dist sales performance graph
export function distributorSalesPerformanceButtonVsOnChange(data) {
    return {
        type: DISTRIBUTOR_SALES_PERFORMANCE_BUTTON_VS,
        data
    };
}


export function distributorSalesPerformanceKpiButtonOnChange(data) {
    return {
        type: DISTRIBUTOR_SALES_PERFORMANCE_KPI_BUTTON,
        data
    };
}

//Product sales performance chart - pareto
export function productPerformanceFetch() {
    return {
        type: PRODUCT_PERFORMANCE_FETCH
    };
}

export function productPerformanceFail() {
    return {
        type: PRODUCT_PERFORMANCE_FAIL
    };
}

export function productPerformanceSuccess(data) {
    return {
        type: PRODUCT_PERFORMANCE_SUCCESS,
        data
    };
}

export function productPerformanceButtonOnChange(data) {
    return {
        type: PRODUCT_PERFORMANCE_BUTTON,
        data
    };
}


//channel perf chart
export function channelPerformanceFetch() {
    return {
        type: CHANNEL_PERFORMANCE_FETCH
    };
}

export function channelPerformanceFail() {
    return {
        type: CHANNEL_PERFORMANCE_FAIL
    };
}

export function channelPerformanceSuccess(data) {
    return {
        type: CHANNEL_PERFORMANCE_SUCCESS,
        data
    };
}
export function channelPerformanceButtonOnChange(data) {
    return {
        type: CHANNEL_PERFORMANCE_BUTTON,
        data
    };
}


//Channel CLICK
export function channelPerformanceClickOnChange(data) {
    return {
        type: CHANNEL_PERFORMANCE_CLICK_ON_CHANGE,
        data
    };
}

//channel filter
export function channelPerformanceFilter(data) {
    return {
        type: CHANNEL_PERFORMANCE_FILTER,
        data
    };
}

//store perf chart
export function storePerformanceFetch(data) {
    return {
        type: STORE_PERFORMANCE_FETCH,
        data
    };
}

export function storePerformanceFail() {
    return {
        type: STORE_PERFORMANCE_FAIL
    };
}

export function storePerformanceSuccess(data) {
    return {
        type: STORE_PERFORMANCE_SUCCESS,
        data
    };
}
export function storePerformanceButtonOnChange(data) {
    return {
        type: STORE_PERFORMANCE_BUTTON,
        data
    };
}
//STORE filter
export function storePerformanceFilter(data) {
    return {
        type: STORE_PERFORMANCE_FILTER,
        data
    };
}

// store table

export function storeTableFetch() {
    return {
        type: STORE_TABLE_FETCH
    };
}

export function storeTableFail() {
    return {
        type: STORE_TABLE_FAIL
    };
}

export function storeTableSuccess(data) {
    return {
        type: STORE_TABLE_SUCCESS,
        data
    };
}
export function storeTableOnChange(data) {
    return {
        type: STORE_TABLE_BUTTON,
        data
    };
}
