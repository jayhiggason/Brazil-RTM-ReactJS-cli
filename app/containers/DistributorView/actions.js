/*
 *
 * DistributorView actions
 *
 */

import {
    DEFAULT_ACTION,
    DISTRIBUTOR_TREND_CHART_DATA_FETCH,
    DISTRIBUTOR_TREND_CHART_DATA_FETCH_FAIL,
    DISTRIBUTOR_TREND_CHART_DATA_FETCH_SUCCESS,
    DISTRIBUTOR_TREND_CHART_ON_FILTER_CHANGE,
    DISTRIBUTOR_TREND_TABLE_DATA_FETCH,
    DISTRIBUTOR_TREND_TABLE_DATA_FETCH_FAIL,
    DISTRIBUTOR_TREND_TABLE_DATA_FETCH_SUCCESS
} from "./constants";


export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

export function distributorTrendChartDataFetch() {
    return {
        type: DISTRIBUTOR_TREND_CHART_DATA_FETCH,
    };
}

export function distributorTrendChartDataFetchSuccess(data) {
    return {
        type: DISTRIBUTOR_TREND_CHART_DATA_FETCH_SUCCESS,
        data
    };
}

export function distributorTrendChartDataFetchFail() {
    return {
        type: DISTRIBUTOR_TREND_CHART_DATA_FETCH_FAIL,
    };
}

export function distributorTrendChartOnFilterChange(data) {
    return {
        type: DISTRIBUTOR_TREND_CHART_ON_FILTER_CHANGE,
        data
    };
}

export function distributorTrendTableDataFetch() {
    return {
        type: DISTRIBUTOR_TREND_TABLE_DATA_FETCH
    };
}

export function distributorTrendTableDataFetchSuccess(data) {
    return {
        type: DISTRIBUTOR_TREND_TABLE_DATA_FETCH_SUCCESS,
        data
    };
}

export function distributorTrendTableDataFetchFail() {
    return {
        type: DISTRIBUTOR_TREND_TABLE_DATA_FETCH_FAIL
    };
}
