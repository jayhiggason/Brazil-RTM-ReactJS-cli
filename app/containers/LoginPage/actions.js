/*
 *
 * LoginPage actions
 *
 */

import {DEFAULT_ACTION, LOGIN_USER, LOGIN_USER_CLOSE_DIALOG, LOGIN_USER_FAIL, LOGIN_USER_SUCCESS,} from "./constants";

export function defaultAction() {
    return {
        type: DEFAULT_ACTION
    };
}

export function loginUser(UserCred) {
    return {
        type: LOGIN_USER,
        UserCred
    }
}


export function loginUserSuccess(data) {
    return {
        type: LOGIN_USER_SUCCESS,
        data
    }
}

export function loginUserFail(message, status) {
    return {
        type: LOGIN_USER_FAIL,
        message: message,
        status: status
    }
}

export function closeGFailedWarningDialog() {
    return {
        type: LOGIN_USER_CLOSE_DIALOG,
    }
}