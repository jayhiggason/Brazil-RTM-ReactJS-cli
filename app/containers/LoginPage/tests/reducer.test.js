// import produce from 'immer';
import loginPageReducer from "../reducer";
import {loginUser} from "../actions";
// import { someAction } from '../actions';

/* eslint-disable default-case, no-param-reassign */
describe("loginPageReducer", () => {
  let state;
  beforeEach(() => {
    state = {
      // default state params here
      userCred:{
        userID:"",
        password:"",
        role:""
      },
      IsAuth: false,
      Loading: false,
      Failed: false,
      FailedMessage:'',
      TokenObj:{}
    }
  });

  it("returns the initial state for reducer", () => {
    const expectedResult = state;
    expect(loginPageReducer(undefined, {})).toEqual(expectedResult);
  });

  it("returns the correct state for user login", () => {
    const UserCred = "data";
    expect(loginPageReducer(state, loginUser(UserCred))).toMatchSnapshot();
  });



  /**
   * Example state change comparison
   *
   * it('should handle the someAction action correctly', () => {
   *   const expectedResult = produce(state, draft => {
   *     draft.loading = true;
   *     draft.error = false;
   *     draft.userData.nested = false;
   *   });
   *
   *   expect(appReducer(state, someAction())).toEqual(expectedResult);
   * });
   */
});
