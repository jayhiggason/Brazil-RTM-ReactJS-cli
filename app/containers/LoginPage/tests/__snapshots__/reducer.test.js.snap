// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginPageReducer returns the correct state for user login 1`] = `
Object {
  "Failed": false,
  "FailedMessage": "",
  "IsAuth": false,
  "Loading": true,
  "TokenObj": Object {},
  "userCred": "data",
}
`;
